src/flamenco/runtime/tests/Local.mk

```diff
@@ -10,6 +11,7 @@ $(call add-objs,fd_exec_sol_compat,fd_flamenco)
 $(call make-unit-test,test_exec_instr,test_exec_instr,fd_flamenco fd_funk fd_ballet fd_util fd_disco,$(SECP256K1_LIBS))
 $(call make-unit-test,test_exec_sol_compat,test_exec_sol_compat,fd_flamenco fd_funk fd_ballet fd_util fd_disco,$(SECP256K1_LIBS))
 $(call make-shared,libfd_exec_sol_compat.so,fd_exec_sol_compat,fd_flamenco fd_funk fd_ballet fd_util fd_disco,$(SECP256K1_LIBS))
+$(call make-shared,libfd_exec_sol_compat_stubbed.so,fd_exec_sol_compat,fd_flamenco fd_funk fd_ballet fd_util fd_disco,$(SECP256K1_LIBS) $(WRAP_FLAGS))
```

Comment:
Why disable fd_execute_instr? Why not just make one binary with everything enabled?

Reply:
But yeah having a second binary is unfortunate, and I'm still not sure if we can mock anything on the Agave end. So this might change.

Reply:
@ravyu-jump Makes sense. Can we move this out to a separate PR? I believe there are better ways to do this, e.g. LD_PRELOADing certain symbols. The wrap hack looks rather non-portable so is likely to break portability.

Reply:
Sure, I can do that. 

> LD_PRELOADing certain symbols
Wouldn't this restrict wrapping to only when using the .so file? IIRC our test-vectors CI runs a statically linked executable instead of the .so so that wouldn't work. But of course open to other ideas

> wrap hack looks rather non-portable
Both gcc and clang support this, unless you mean architecture-wise?

Reply:
> Wouldn't this restrict wrapping to only when using the .so file? IIRC our test-vectors CI runs a statically linked executable instead of the .so so that wouldn't work.

The wrap hack is currently only applied to the shared library which is why I suggested LD_PRELOAD instead.
If we need this for test executables too why not just fix this in the source code? We could trivially indirect the instruction executor call.

> Both gcc and clang support this, unless you mean architecture-wise?

It's a link option, so must be supported by bfd, ld.lld, mold, etc. On most systems the linker is provided by the system and you can't replace it as easily as the compiler.

Reply:
As most users don't actually need this binary, maybe we can conditionally build this for the few systems that the core devs use? To avoid breaking the whole build.

Reply:
I agree @topointon-jump, but the wrap linker hack is not the right way to achieve this.

Reply:
Moved this to a separate PR: https://github.com/firedancer-io/firedancer/pull/2601